internal/pkg/template/templates/environment/cf.yml

```diff
@@ -37,14 +37,23 @@ Conditions:
   DelegateDNS:
     !Not [!Equals [ !Ref AppDNSName, "" ]]
   ExportHTTPSListener: !And
-    - !Condition DelegateDNS
     - !Condition CreateALB
+    - !Condition UseHTTPS
+  UseHTTPS: !Or
+    - !Condition DelegateDNS
+    - !Condition HasImportedCert
   CreateEFS:
     !Not [!Equals [ !Ref EFSWorkloads, ""]]
   CreateNATGateways:
     !Not [!Equals [ !Ref NATWorkloads, ""]]
   HasAliases:
     !Not [!Equals [ !Ref Aliases, "" ]]
+  HasImportedCert:
+{{- if .ImportCertARNs}}
+    !Equals [ 'true', 'true' ]
```

Comment:
What do you think of having the cert ARNs as a parameter instead? and then doing `!Not [!Equals [ !Ref PublicALBCerts, "" ]]`

I think we have basically two options:
1. Either use CFN conditions properly with parameters and no Go templates
2. We delete the `ExportHTTPSListener` and `HasImportedCert` conditions and replace them with go template logic `{{if .NeedsToExportHTTPSListener }}`

Reply:
We can't do 1 because CFN doesn't support looping. 2 sounds ok to me. It's just I find too much go template make the template more difficult to read.

Alternatively, we could have both import cert as go template and CFN, and we pass the same value for them. That way we refer to the CFN param as much as we can, and for looping we use go template.